enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,416 @@
+* **kmod-{vendor}-src.rpm** The next step is a source RPM package that can be recompiled with rpmbuild. This is also the base for akmods, dkms
+* **precompiled-{vendor}-{kernelversion}.rpm** Precompiled RPMS this is the wish thinking to have in the future. DriverContainers could be easily built just by using RPMs. 
+
+Some drivers will *never* be upstreamed and can be in any of the states described above.  The proposed solutions needs to handle drivers in any state and build by "any" tool. 
+
+The compilation of kernel modules was always anticipated to be the fallback solution when dealing with kernel modules. Some kernel modules will always be out-of-tree and are not going to be included in the near future. Some kernel modules are out-of-tree but we are working with the vendors on upstreaming them  to the mainline kernel. 
+This work is being done by a virtual-team consisting of an  OpenShift-accelerators  and RHEL-accelerator team responsible for enabling hardware and software on RHEL and OpenShift. 
+
+
+### Goals
+
+
+* A unified way to deploy out-of-tree drivers on OpenShift 4.x on all supported Red Hat Operating Systems
+* The solution should avoid rebuilds on every node and allow for distribution of drivers on a cluster using the cluster registry
+* A solution for day-1 and day-2 kernel modules
+* Support upgrades of OpenShift Support of multiple kernel module providers
+* Hierarchical initialization of kernel modules (modeling dependencies)
+*  Handle dependencies between kernel modules (depmod) in tree and out of tree
+* Should support disconnected and proxy environments
+* Support building a DriverContainer on any host or kernel version (-k flags) e.g CI
+* Support heterogeneous cluster: 
+  * OpenShift with RHEL7, 8 and RHCOS 
+  * OKD is only supporting FCOS nodes for now. 
+* Extended Update Support
+
+
+### Non-Goals
+
+* The solution is not a replacement for the traditional way of delivering kernel modules, customer/partners should be aware that we prefer they  deliver the drivers upstream
+* We are not providing a way how to build the drivers this is business logic of a specific vendor, we are providing the interface to hook into specific stages of a DriverContainer
+
+## Proposal
+
+The SRO pattern showed how to enable hardware and the complete hardware accelerator stack on OpenShift. The heavy-lifting was the management of the DriverContainer. Approximately 5% of the logic behind SRO was used for deploying the remaining parts aka stack. 
+
+Based on SRO we are going to create a new operator that will solely be used for DriverContainer management providing out of tree drivers on OpenShift. 
+
+SRO can be seen as the upstream project where we test new features and enable more hardware accelerators where this new operator will be the downstream operator shipped by OpenShift for out-of-tree drivers. 
```

Comment:
shipped as optional building block to consume from OLM?

Reply:
SRO is already a community-operator OLM, the "new" (downstream) operator is anticipated to be part of the payload.

Reply:
I think this is one thing where we want to make a call whether it makes sense to ship this by default to simplify life for users since the intent is that one operator will support multiple drivers.